crates/http/src/ext.rs

```diff
@@ -89,17 +89,25 @@ pub trait ServiceExt<Body>: Sized {
         FormUrlencodedRequest::new(self)
     }
 
-    fn catch_http_code<M>(self, status_code: StatusCode, mapper: M) -> CatchHttpCodes<Self, M>
+    /// Catches responses with the given status code and then maps those
+    /// responses to an error type using the provided `mapper` function.
+    fn catch_http_code<M, ResBody, E>(
+        self,
+        status_code: StatusCode,
+        mapper: M,
+    ) -> CatchHttpCodes<Self, M>
     where
-        M: Clone,
+        M: Clone + Fn(Response<ResBody>) -> E + Send + Clone + 'static,
```

Comment:
`Clone` is redundant here, `Fn(Response<ResBody>) -> E` is one trait bound, the list after are other traits associated to `M`, not to the function result

If we wanted to bind traits on the result, we would have bound them to `E`